src/bot/commands/quotes.ts

```diff
@@ -0,0 +1,130 @@
+import Discord from 'discord.js';
+import MongoService from 'src/core/services/mongo.service';
+import ICommand from './ICommand';
+
+interface Quote {
+  author: string;
+  quote: string;
+  meta: {
+    authorCachedName: string;
+    createdBy: string;
+    createdByCachedName: string;
+    createdAt: Date;
+  };
+}
+
+const command: ICommand = {
+  name: 'quotes',
+  aliases: ['quote', 'quotes', 'q'],
+  description: 'Get random quotes from people in chat, or add quotes to the list.',
+  async execute(message, args, _prefix, _commands, db) {
+    // Get random quote
+    if (args.filter((s) => s.trim().length).length === 0) {
+      getRandomQuote(message, args, db);
+      return;
+    }
+
+    const first = args[0].trim().toLowerCase();
+
+    switch (first) {
+      // Search quotes
+      case 'search':
+        searchQuotes(message, args.slice(1), db);
+        return;
+
+      // Add quote
+      case 'add':
+      default:
+        addNewQuote(message, args.slice(first === 'add' ? 1 : 0), db);
+    }
+  },
+};
+
+const clean = (str: string): string => str.replace(/[\t\n|]+/g, ' ').replace(/\s+/g, ' ');
+const getQuoteStr = ({ author, quote }: Quote): string => `"${quote}" - ${author}`;
+
+async function getRandomQuote(message: Discord.Message, args: string[], db: MongoService): Promise<void> {
+  const count = await db.count(message.author.id, 'quotes', {});
+  const r = Math.floor(Math.random() * count);
+  const q = await db.dbInstance.collection('quotes').find().skip(r).limit(1).toArray();
+
+  if (q.length > 0) {
+    const quote: Quote = q[0];
+    message.reply(getQuoteStr(quote));
+  } else {
+    message.reply("This is where I would usually put a quote. I can't remember any, for some reason...");
+  }
+}
+
+async function searchQuotes(message: Discord.Message, args: string[], db: MongoService): Promise<void> {
+  const q = await db.find<Quote>(message.author.id, 'quotes', {
+    quote: {
+      $regex: `${args.join(' ')}`,
+      $options: 'i',
+    },
+  });
+
+  if (q.length > 0) {
+    message.reply("Found a few, I'll DM you what I got!");
+    message.author.send(`Found ${q.length} quote${q.length !== 1 ? 's' : ''}:`);
+    q.forEach((quote) => {
+      message.author.send(getQuoteStr(quote));
```

Comment:
Rather than sending messages within a loop can we append the strings to an array with new line characters and send the array as one? It's how the help command does it, would be more efficient to only send the one.

Reply:
Sure, I'll update that.